src/main.js

```diff
@@ -35,64 +35,83 @@ const storageApi = {};
 /**
  * Encapsulates authentication flow logic.
  * @param {Object} options Options object.
- * @param {string} options.apiKey The firebase API key
+ * @param {string} options.apiKey The Firebase API key.
  * @param {string} options.redirectUri The redirect URL used by OAuth providers.
  * @param {Array.<ProviderOptions|string>} options.providers Array of arguments that will be passed to the addProvider method.
  */
 export default class Auth {
-	constructor({ name = 'default', apiKey, redirectUri, storage = storageApi } = {}) {
-		if (typeof apiKey !== 'string') throw Error('The argument "apiKey" is required');
-
-		Object.assign(this, {
-			name,
-			apiKey,
-			storage,
-			redirectUri,
-			listeners: []
-		});
-
-		this.storage.get(this.sKey('User')).then(user => {
-			this.setState(JSON.parse(user), false);
-			if (this.user)
-				this.refreshIdToken()
-					.then(() => this.fetchProfile())
-					.catch(e => {
-						if (e.message === 'TOKEN_EXPIRED' || e.message === 'INVALID_ID_TOKEN') return this.signOut();
-						throw e;
-					});
-		});
+	constructor({
+		apiKey,
+		redirectUri,
+		name = 'default',
+		storage = storageApi,
+	} = {}) {
+		if (!apiKey) {
+			throw Error('The argument "apiKey" is required');
+		} else {
+			Object.assign(this, {
+				apiKey,
+				redirectUri,
+				name,
+				storage,
+				listeners: [],
+			});
 
-		// Because this library is used in react native, outside the browser as well,
-		// we need to first check if this environment supports `addEventListener` on the window.
-		'addEventListener' in window &&
-			window.addEventListener('storage', e => {
-				// This code will run if localStorage for this user
-				// data was updated from a different browser window.
-				if (e.key !== this.sKey('User')) return;
-				this.setState(JSON.parse(e.newValue), false);
+			this.storage.get(this.sKey('User')).then(user => {
+				this.setState(JSON.parse(user), false);
+				if (this.user) {
+					this.refreshIdToken()
+						.then(() => this.fetchProfile())
+						.catch(error => {
+							switch (error.message) {
+								case 'TOKEN_EXPIRED':
```

Comment:
Switch takes much more space than the other way. This code is aggressively optimized for space. Take that into account. It was not meant to be read by the average developer.